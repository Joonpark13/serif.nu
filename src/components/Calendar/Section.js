import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import { getDurationInHours, getFormattedClassEvent } from 'util/time';

export const MAX_WIDTH_PERCENT = 97;

export const styles = {
  paper: {
    position: 'absolute',
    top: ({ section }) => {
      const minute = section.event.start.minute;
      const offset = Math.round(minute / 60 * 100);
      return `${offset}%`;
    },
    left: ({ section }) => `${MAX_WIDTH_PERCENT * section.columnWidth * section.column}%`,
    height: ({ section }) => {
      const durationInHours = getDurationInHours(section.event);
      const heightInPercent = Math.round(durationInHours * 100);
      return `${heightInPercent}%`;
    },
    width: ({ section }) => `${MAX_WIDTH_PERCENT * section.columnWidth}%`,
    backgroundColor: ({ section }) => section.color,
    overflow: 'hidden',
  },
  container: {
    margin: '3px',
  },
  text: {
    color: 'white',
  },
  header: {
    display: 'flex',
    justifyContent: 'space-between',
    flexWrap: 'wrap',
  },
};

function Section({ classes, section }) {
  return (
    <Paper className={classes.paper}>
      <div className={classes.container}>
        <div className={classes.header}>
          <Typography variant="caption" className={classes.text}>
            {getFormattedClassEvent(section.event)}
          </Typography>

          <Typography variant="caption" className={classes.text}>
            {`${section.subjectId} ${section.courseId}`}
          </Typography>
        </div>

        <Typography variant="subtitle2" className={`${classes.text} ${classes.name}`} noWrap>
          {section.name}
        </Typography>
      </div>
    </Paper>
  );
}

Section.propTypes = {
  section: PropTypes.objectOf(PropTypes.any).isRequired, // TODO
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export { Section as UnstyledSection };
export default withStyles(styles)(Section);
