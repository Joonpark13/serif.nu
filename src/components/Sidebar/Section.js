import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Typography from '@material-ui/core/Typography';
import { getFormattedClassSchedule } from 'util/time';

export const styles = {
  sectionTitle: {
    fontWeight: 'bold',
  },
};

function Section({ addSection, section, disabled, classes }) {
  return (
    <ListItem
      key={section.id}
      button
      onClick={() => addSection(section)}
      disabled={disabled}
    >
      <ListItemText>
        <Typography variant="h6" className={classes.sectionTitle}>
          {`Section ${section.sectionNumber}`}
        </Typography>

        {section.topic && <Typography>{section.topic}</Typography>}

        {section.schedule.map((scheduleObj, index) => (
          /* eslint-disable react/no-array-index-key */
          <Typography key={index}>
            {getFormattedClassSchedule(scheduleObj)}
          </Typography>
          /* eslint-enable react/no-array-index-key */
        ))}

        <Typography>{section.schedule[0].location}</Typography>

        <Typography>
          {section.instructors.map((teacher, idx) => (
            `${teacher}${idx !== section.instructors.length - 1 ? ', ' : ''}`))
          }
        </Typography>
      </ListItemText>
    </ListItem>
  );
}

Section.propTypes = {
  addSection: PropTypes.func.isRequired,
  section: PropTypes.objectOf(PropTypes.any).isRequired, // TODO
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  disabled: PropTypes.bool,
};

Section.defaultProps = {
  disabled: false,
};

export { Section as UnstyledSection };
export default withStyles(styles)(Section);
