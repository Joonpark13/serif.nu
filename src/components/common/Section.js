import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

function Section({ classes, leftHeaderContent, rightHeaderContent, sectionName, onClick }) {
  return (
    <Paper elevation={2} className={classes.paper} onClick={onClick}>
      <div className={classes.container}>
        <div className={classes.header}>
          <Typography variant="caption" className={classes.text}>
            {leftHeaderContent}
          </Typography>

          <Typography variant="caption" className={classes.text}>
            {rightHeaderContent}
          </Typography>
        </div>

        <Typography variant="subtitle2" className={`${classes.text} ${classes.name}`} noWrap>
          {sectionName}
        </Typography>
      </div>
    </Paper>
  );
}

Section.propTypes = {
  classes: PropTypes.shape({
    paper: PropTypes.string,
    container: PropTypes.string,
    header: PropTypes.string,
    text: PropTypes.string,
    name: PropTypes.string,
  }).isRequired,
  leftHeaderContent: PropTypes.string,
  rightHeaderContent: PropTypes.string,
  sectionName: PropTypes.string,
  onClick: PropTypes.func,
};

Section.defaultProps = {
  leftHeaderContent: '',
  rightHeaderContent: '',
  sectionName: '',
  onClick: () => {},
};

export default Section;
