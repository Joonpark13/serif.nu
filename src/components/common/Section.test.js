import React from 'react';
import { shallow } from 'enzyme';
import Section from './Section';

describe('Section', () => {
  const classes = {
    paper: '',
    container: '',
    header: '',
    text: '',
    name: '',
  };

  it('renders correctly', () => {
    const wrapper = shallow(
      <Section
        classes={classes}
        leftHeaderContent="Left"
        rightHeaderContent="Right"
        sectionName="Intro to Whatever"
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should verify onClickFn as a default prop', () => {
    expect(Section.defaultProps.onClickFn()).not.toBeDefined();
  });
});
