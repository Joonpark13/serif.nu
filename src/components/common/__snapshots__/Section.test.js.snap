// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Section
    classes={
      Object {
        "container": "",
        "header": "",
        "name": "",
        "paper": "",
        "text": "",
      }
    }
    leftHeaderContent="Left"
    onClickFn={[Function]}
    rightHeaderContent="Right"
    sectionName="Intro to Whatever"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className=""
      >
        <div
          className=""
        >
          <WithStyles(Typography)
            className=""
            variant="caption"
          >
            Left
          </WithStyles(Typography)>
          <WithStyles(Typography)
            className=""
            variant="caption"
          >
            Right
          </WithStyles(Typography)>
        </div>
        <WithStyles(Typography)
          className=" "
          noWrap={true}
          variant="subtitle2"
        >
          Intro to Whatever
        </WithStyles(Typography)>
      </div>,
      "className": "",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className=""
          >
            <WithStyles(Typography)
              className=""
              variant="caption"
            >
              Left
            </WithStyles(Typography)>
            <WithStyles(Typography)
              className=""
              variant="caption"
            >
              Right
            </WithStyles(Typography)>
          </div>,
          <WithStyles(Typography)
            className=" "
            noWrap={true}
            variant="subtitle2"
          >
            Intro to Whatever
          </WithStyles(Typography)>,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                className=""
                variant="caption"
              >
                Left
              </WithStyles(Typography)>,
              <WithStyles(Typography)
                className=""
                variant="caption"
              >
                Right
              </WithStyles(Typography)>,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Left",
                "className": "",
                "variant": "caption",
              },
              "ref": null,
              "rendered": "Left",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Right",
                "className": "",
                "variant": "caption",
              },
              "ref": null,
              "rendered": "Right",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Intro to Whatever",
            "className": " ",
            "noWrap": true,
            "variant": "subtitle2",
          },
          "ref": null,
          "rendered": "Intro to Whatever",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className=""
        >
          <div
            className=""
          >
            <WithStyles(Typography)
              className=""
              variant="caption"
            >
              Left
            </WithStyles(Typography)>
            <WithStyles(Typography)
              className=""
              variant="caption"
            >
              Right
            </WithStyles(Typography)>
          </div>
          <WithStyles(Typography)
            className=" "
            noWrap={true}
            variant="subtitle2"
          >
            Intro to Whatever
          </WithStyles(Typography)>
        </div>,
        "className": "",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className=""
            >
              <WithStyles(Typography)
                className=""
                variant="caption"
              >
                Left
              </WithStyles(Typography)>
              <WithStyles(Typography)
                className=""
                variant="caption"
              >
                Right
              </WithStyles(Typography)>
            </div>,
            <WithStyles(Typography)
              className=" "
              noWrap={true}
              variant="subtitle2"
            >
              Intro to Whatever
            </WithStyles(Typography)>,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <WithStyles(Typography)
                  className=""
                  variant="caption"
                >
                  Left
                </WithStyles(Typography)>,
                <WithStyles(Typography)
                  className=""
                  variant="caption"
                >
                  Right
                </WithStyles(Typography)>,
              ],
              "className": "",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Left",
                  "className": "",
                  "variant": "caption",
                },
                "ref": null,
                "rendered": "Left",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Right",
                  "className": "",
                  "variant": "caption",
                },
                "ref": null,
                "rendered": "Right",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Intro to Whatever",
              "className": " ",
              "noWrap": true,
              "variant": "subtitle2",
            },
            "ref": null,
            "rendered": "Intro to Whatever",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
